devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -227,7 +227,8 @@ public boolean isSatisfiedBy(Task t) {
 
         TaskProvider<QuarkusDev> quarkusDev = tasks.register(QUARKUS_DEV_TASK_NAME, QuarkusDev.class, devRuntimeDependencies,
                 quarkusExt);
-        TaskProvider<QuarkusRun> quarkusRun = tasks.register(QUARKUS_RUN_TASK_NAME, QuarkusRun.class);
+        TaskProvider<QuarkusRun> quarkusRun = tasks.register(QUARKUS_RUN_TASK_NAME, QuarkusRun.class,
+                build -> build.dependsOn(quarkusBuild));
```

Comment:
Should `quarkusDev` depend on `quarkusBuild` too?

Reply:
No, `quarkusBuild` produces a packaged app, which isn't necessary for `quarkusDev`.